espnet2/bin/tts_decode.py

```diff
@@ -129,46 +164,44 @@ def tts_decode(
 
             # TODO(kamo): Use common gathering attention system and plot it
             #  Now att_ws is not used.
-
-            with torch.no_grad():
-                outs, probs, att_ws = tts.inference(
-                    **_data,
-                    threshold=threshold,
-                    maxlenratio=maxlenratio,
-                    minlenratio=minlenratio,
-                )
-                if normalize is not None:
-                    outs, _ = normalize.inverse(outs[None])
-                outs = outs.squeeze(0)
-
+            outs, probs, att_ws = tts.inference(
+                **_data,
+                threshold=threshold,
+                maxlenratio=maxlenratio,
+                minlenratio=minlenratio,
+            )
+            outs_denorm = normalize.inverse(outs[None])[0][0]
             insize = next(iter(_data.values())).size(0)
             logging.info(
                 "inference speed = {} msec / frame.".format(
-                    (time.perf_counter() - start_time)
-                    / (int(outs.size(0)) * 1000)
+                    (time.perf_counter() - start_time) / (int(outs.size(0)) * 1000)
                 )
             )
             if outs.size(0) == insize * maxlenratio:
                 logging.warning(
                     f"output length reaches maximum length ({key})."
                 )
-
             logging.info(
                 f"({idx}/{len(batch_sampler)}) {key} "
                 f"(size:{insize}->{outs.size(0)})"
             )
             f[key] = outs.cpu().numpy()
+            g[key] = outs_denorm.cpu().numpy()
+
+            wav = spectrogram2wav(outs_denorm.cpu().numpy(), **spectrogram2wav_params)
+            sf.write(f"{output_dir}/wav/{key}.wav", wav, spectrogram2wav_params["fs"], "PCM_16")
```

Comment:
I think wav scp file is useful for after processing, e.g. ASR-evaluation.
https://github.com/espnet/espnet/blob/f423c25e7081b4728064f494d2f2548611dea149/espnet2/utils/fileio.py#L218-L232
This class can generates wave files with scp text as the kaldi writer.

Reply:
This is nice. I will introduce it.

Reply:
`wav.scp` is useful for ASR evaluation but wav files are separately saved. TTS users may want to access the directory including all of wav files. For now, I did not come up with a good idea to achieve both wishes, so I keep current style.